tests/test_rest.py

```diff
@@ -759,3 +772,211 @@ def on_service_state_change(zeroconf: Zeroconf, service_type: str, name: str, st
   # check advertisememts
   assert AMPLIPI_ZC_NAME in services_advertised
   assert services_advertised[AMPLIPI_ZC_NAME].port == 9898
+
```

Comment:
These tests are really valuable! I know they are pretty verbose but I don't know how much is gained by refactoring.